reinforcement_learning/rlclientlib/logger/event_batcher.cc

```diff
@@ -0,0 +1,53 @@
+#include "event_batcher.h"
+
+#include "generated/RankingEvent_generated.h"
+#include "generated/OutcomeEvent_generated.h"
+
+namespace reinforcement_learning {
+  namespace utility {
+
+    event_batcher::event_batcher()
+    {
+    }
+
+    void event_batcher::batch_serialize(data_buffer& oss, size_t& remaining, event_queue<ranking_event>& queue, size_t _send_high_water_mark)
+    {
+      batch_serialize_internal<ranking_event, RankingEvent, VW::Events::RankingEventBatchBuilder>(oss, remaining, queue, _send_high_water_mark, _batch_size);
+    }
+    void event_batcher::batch_serialize(data_buffer& oss, size_t& remaining, event_queue<outcome_event>& queue, size_t _send_high_water_mark)
+    {
+      batch_serialize_internal<outcome_event, OutcomeEvent, VW::Events::OutcomeEventBatchBuilder>(oss, remaining, queue, _send_high_water_mark, _batch_size);
+    }
+
+    template<typename TEvent, typename TResultObject, typename TResultObjectBuilder>
+    void event_batcher::batch_serialize_internal(data_buffer& oss, size_t& remaining, event_queue<TEvent>& queue, size_t _send_high_water_mark, size_t batch_size)
```

Comment:
since _batch_size is the member of the class, we do not need it here as input argument.
The same has to be applied to send_high_water_mark (and it has to be removed from async_batcher since this is the guy who owns batching logic)

Reply:
We will need that. The logic right now is using batch_size to determine

1. with a maximum of 400 events will it fill high_water_mark? 
2. If yes, try a recursive call with half of the value. That is why batch_size is needed in the signature.


---
In reply to: [228620136](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1653#discussion_r228620136) [](ancestors = 228620136)